### PR TITLE
Fixup for `source.name` change detection

### PR DESCRIPTION
**Pull request by crisptrutski**
This fixes false positives on the guard, which has a large performance impact.

### CODE REVIEW

**Comment by johnjbarton** [comment]
Can we come up with a test that shows the problem and prevents a re-occurrence?

**Comment by crisptrutski** [comment]
The only test I think we could write would require using a test spy, as the body of the `if` is idempotent. Not sure if this is worth adding a library like Sinon for, or if I should make an adhoc spy.

In case the issue isn't self evident enough: the `if` statement will ALWAYS be true if `this.sourceRoot_` is defined.

File: src/outputgeneration/ParseTreeMapWriter.js [hunk 1]
```diff
@@ -114,7 +114,8 @@ export class ParseTreeMapWriter extends ParseTreeWriter {
       line: line,
       column: position.column || 0  // source map uses zero based columns
     };
-    if (position.source.name !== this.sourceName_) {
+    if (position.source.name !== this.rawSourceName_) {
```
Comment (johnjbarton) [inline src/outputgeneration/ParseTreeMapWriter.js]: Please rename `this.sourceName_` to `this.relativeSourceName_` and use `this.sourceName_` for your fix.

Comment (crisptrutski) [inline src/outputgeneration/ParseTreeMapWriter.js]: sure

**Comment by johnjbarton** [comment]
Ok now I see that the cache value was changed in b8eab4f to be a value that could never match the input value.

**Comment by johnjbarton** [comment]
Sorry, I think you were correct about the test. Shall we omit it? Your call, just let me know.

**Comment by crisptrutski** [comment]
Yeah, think we should let the test go (too brittle). Assume you're cherry-picking again so I can leave omission to you? :)

**Comment by johnjbarton** [comment]
I use `pulley` which squashes and uses the PR top comment as the commit message.

**Comment by crisptrutski** [comment]
OK, I'll push over this branch quick

**Comment by crisptrutski** [comment]
@johnjbarton test retracted

File: src/outputgeneration/ParseTreeMapWriter.js [hunk 2]
```diff
@@ -115,7 +115,8 @@ export class ParseTreeMapWriter extends ParseTreeWriter {
       column: position.column || 0  // source map uses zero based columns
     };
     if (position.source.name !== this.sourceName_) {
-      this.sourceName_ = relativeToSourceRoot(position.source.name,
+      this.sourceName_ = position.source.name;
+      this.relativeSourceName_ = relativeToSourceRoot(position.source.name,
```
Comment (arv) [inline src/outputgeneration/ParseTreeMapWriter.js]: `relativeSourceName_` should have been added to the constructor to keep the shape of the object.

Comment (johnjbarton) [inline src/outputgeneration/ParseTreeMapWriter.js]: yes, and `this.sourceName_` as well, though in this case we have almost singleton compared to the work to build the sourcemap.

**Pull request closed without merging**